kernel/packages/shared/comms/index.ts

```diff
@@ -681,7 +681,7 @@ export async function connect(userId: string) {
         connectionAnalytics.visiblePeers = context?.stats.visiblePeerIds.map(it => it.slice(-6))
 
         if (connectionAnalytics) {
-          queueTrackingEvent('Comms status', connectionAnalytics)
+          queueTrackingEvent('New Comms Status', connectionAnalytics)
```

Comment:
@dnul Quick question. The format of the analytics that we were sending has changed a bit. What would be the approach for adding those changes? is OK to change the event? Or can we use the same event and get away with it?

Reply:
@pablitar  safest would be a new event if we're removing any fields. Let me know when pushed this events to dev and I'll check them out. thanks

Reply:
@dnul I've been sending a few events. This is an example:
```
{
    "connectedPeers": [
      "dfb47f",
      "939f7c",
      "838551",
      "7df8fa",
      "bd27ae",
      "80d02b",
      "26ef82"
    ],
    "stats": {
      "periodLength": 1000,
      "sent": {
        "expired": 0,
        "expiredPercentage": 0,
        "packetDuplicates": 0,
        "duplicatePercentage": 0,
        "averagePacketSize": 492.3470157700143,
        "optimistic": 13233,
        "packets": 13253,
        "totalBytes": 6525075,
        "lastPeriodUpdate": 267068.11500000185,
        "_bytesPerSecond": {
          "accumulatedInPeriod": 3528,
          "currentValue": 108700.15147635363
        },
        "_packetsPerSecond": {
          "accumulatedInPeriod": 7,
          "currentValue": 181.97179437159193
        },
        "_expiredPerSecond": {
          "accumulatedInPeriod": 0,
          "currentValue": 0
        },
        "_duplicatesPerSecond": {
          "accumulatedInPeriod": 0,
          "currentValue": 0
        },
        "statsByType": {
          "position": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 425.3694212849436,
            "optimistic": 8156,
            "packets": 8156,
            "totalBytes": 3469313,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 34434.662627239704
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 69.98915168138151
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "profile": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 439.0723488602577,
            "optimistic": 1009,
            "packets": 1009,
            "totalBytes": 443024,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 3528,
              "currentValue": 3527.453244741628
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 7,
              "currentValue": 6.998915168138151
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "chat": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 213.25,
            "optimistic": 0,
            "packets": 20,
            "totalBytes": 4265,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "parcelUpdate": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 475.2520325203252,
            "optimistic": 123,
            "packets": 123,
            "totalBytes": 58456,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "sceneComms": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 646.3921419518377,
            "optimistic": 3945,
            "packets": 3945,
            "totalBytes": 2550017,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 70738.03560437229
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 104.98372752207226
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          }
        }
      },
      "received": {
        "expired": 8973,
        "expiredPercentage": 0.24165140579554023,
        "packetDuplicates": 1929,
        "duplicatePercentage": 0.05194980071097705,
        "averagePacketSize": 656.5058170849941,
        "optimistic": 37100,
        "packets": 37132,
        "totalBytes": 24377374,
        "lastPeriodUpdate": 267068.11500000185,
        "_bytesPerSecond": {
          "accumulatedInPeriod": 565,
          "currentValue": 124218.74609416394
        },
        "_packetsPerSecond": {
          "accumulatedInPeriod": 1,
          "currentValue": 189.9705545637498
        },
        "_expiredPerSecond": {
          "accumulatedInPeriod": 1,
          "currentValue": 45.99287110490785
        },
        "_duplicatesPerSecond": {
          "accumulatedInPeriod": 0,
          "currentValue": 0.9998450240197359
        },
        "statsByType": {
          "profile": {
            "expired": 785,
            "expiredPercentage": 0.09504782661339145,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 686.7470638091779,
            "optimistic": 8259,
            "packets": 8259,
            "totalBytes": 5671844,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 25485.049817239047
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 35.99442086471049
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "position": {
            "expired": 8159,
            "expiredPercentage": 0.3415522438044206,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 632.5580626255861,
            "optimistic": 23888,
            "packets": 23888,
            "totalBytes": 15110547,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 565,
              "currentValue": 86408.6066658336
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 1,
              "currentValue": 133.9792332186446
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 1,
              "currentValue": 45.99287110490785
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "sceneComms": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 1903,
            "duplicatePercentage": 0.4278327338129496,
            "averagePacketSize": 728.4249100719425,
            "optimistic": 4448,
            "packets": 4448,
            "totalBytes": 3240034,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 11609.200573893153
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 18.99705545637498
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0.9998450240197359
            }
          },
          "parcelUpdate": {
            "expired": 29,
            "expiredPercentage": 0.05742574257425743,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 684.7841584158416,
            "optimistic": 505,
            "packets": 505,
            "totalBytes": 345816,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 715.8890371981308
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0.9998450240197359
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "chat": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 26,
            "duplicatePercentage": 0.8125,
            "averagePacketSize": 285.40625,
            "optimistic": 0,
            "packets": 32,
            "totalBytes": 9133,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          }
        }
      },
      "relayed": {
        "expired": 0,
        "expiredPercentage": 0,
        "packetDuplicates": 0,
        "duplicatePercentage": 0,
        "averagePacketSize": 717.9429993577393,
        "optimistic": 37347,
        "packets": 37368,
        "totalBytes": 26828094,
        "lastPeriodUpdate": 267068.11500000185,
        "_bytesPerSecond": {
          "accumulatedInPeriod": 0,
          "currentValue": 280848.46848695166
        },
        "_packetsPerSecond": {
          "accumulatedInPeriod": 0,
          "currentValue": 371.9423489353417
        },
        "_expiredPerSecond": {
          "accumulatedInPeriod": 0,
          "currentValue": 0
        },
        "_duplicatesPerSecond": {
          "accumulatedInPeriod": 0,
          "currentValue": 0
        },
        "statsByType": {
          "position": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 695.76337789094,
            "optimistic": 23565,
            "packets": 23565,
            "totalBytes": 16395664,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 159688.24832126408
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 219.9659052843419
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "profile": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 741.7707679795516,
            "optimistic": 8607,
            "packets": 8607,
            "totalBytes": 6384421,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 46363.81360881917
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 60.990546465203884
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "parcelUpdate": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 755.072,
            "optimistic": 500,
            "packets": 500,
            "totalBytes": 377536,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 2543.605741106208
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 2.9995350720592073
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "sceneComms": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 783.8008556149732,
            "optimistic": 4675,
            "packets": 4675,
            "totalBytes": 3664269,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 72252.8008157622
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 87.98636211373676
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "chat": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 295.42857142857144,
            "optimistic": 0,
            "packets": 21,
            "totalBytes": 6204,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          }
        }
      },
      "all": {
        "expired": 8973,
        "expiredPercentage": 0.10225291443027588,
        "packetDuplicates": 1929,
        "duplicatePercentage": 0.021982154456257907,
        "averagePacketSize": 657.8754344580813,
        "optimistic": 87680,
        "packets": 87753,
        "totalBytes": 57730543,
        "lastPeriodUpdate": 267068.11500000185,
        "_bytesPerSecond": {
          "accumulatedInPeriod": 4093,
          "currentValue": 513767.3660574692
        },
        "_packetsPerSecond": {
          "accumulatedInPeriod": 8,
          "currentValue": 743.8846978706835
        },
        "_expiredPerSecond": {
          "accumulatedInPeriod": 1,
          "currentValue": 45.99287110490785
        },
        "_duplicatesPerSecond": {
          "accumulatedInPeriod": 0,
          "currentValue": 0.9998450240197359
        },
        "statsByType": {
          "profile": {
            "expired": 785,
            "expiredPercentage": 0.04391608391608392,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 699.2609230769231,
            "optimistic": 17875,
            "packets": 17875,
            "totalBytes": 12499289,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 3528,
              "currentValue": 75376.31667079985
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 7,
              "currentValue": 103.98388249805252
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "position": {
            "expired": 8159,
            "expiredPercentage": 0.14672085453793451,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 628.9543778884713,
            "optimistic": 55609,
            "packets": 55609,
            "totalBytes": 34975524,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 565,
              "currentValue": 280531.5176143374
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 1,
              "currentValue": 423.934290184368
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 1,
              "currentValue": 45.99287110490785
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "sceneComms": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 1903,
            "duplicatePercentage": 0.14562289562289563,
            "averagePacketSize": 723.4710743801653,
            "optimistic": 13068,
            "packets": 13068,
            "totalBytes": 9454320,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 154600.03699402764
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 211.967145092184
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0.9998450240197359
            }
          },
          "parcelUpdate": {
            "expired": 29,
            "expiredPercentage": 0.025709219858156027,
            "packetDuplicates": 0,
            "duplicatePercentage": 0,
            "averagePacketSize": 693.0921985815603,
            "optimistic": 1128,
            "packets": 1128,
            "totalBytes": 781808,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 3259.494778304339
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 3.9993800960789434
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          },
          "chat": {
            "expired": 0,
            "expiredPercentage": 0,
            "packetDuplicates": 26,
            "duplicatePercentage": 0.3561643835616438,
            "averagePacketSize": 268.52054794520546,
            "optimistic": 0,
            "packets": 73,
            "totalBytes": 19602,
            "lastPeriodUpdate": 267068.11500000185,
            "_bytesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_packetsPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_expiredPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            },
            "_duplicatesPerSecond": {
              "accumulatedInPeriod": 0,
              "currentValue": 0
            }
          }
        }
      },
      "periodId": 32778
    },
    "trackedPeers": [
      "936ffa"
    ],
    "visiblePeers": [
      
    ],
    "time": "2020-04-07T14:10:23.180Z",
    "sessionId": "4e6e777d-6165-4420-a9a1-dfd7c488a997"
  }
```

Reply:
@dnul does segment manage these nested structures correctly?

Reply:
@dnul I've updated the event to be consistent with the performance tests we are doing. It now looks something like this:

Please let me know if this would be OK!

```
{
  "...": "...",
  "properties": {
    "connectedPeers": [
      "39b4e0",
      "9650a4",
      "8a0fab",
      "8c39a6"
    ],
    "stats": {
      "sent": 40,
      "sentTotal": 484,
      "sentPerSecond": 39.992601368715455,
      "sentBytes": 13868,
      "sentTotalBytes": 164361,
      "sentBytesPerSecond": 13865.434894533648,
      "sentAveragePacketSize": 339.5888429752066,
      "received": 195,
      "receivedTotal": 5154,
      "receivedPerSecond": 194.96393167248783,
      "receivedBytes": 125089,
      "receivedTotalBytes": 3345703,
      "receivedBytesPerSecond": 125065.86281528119,
      "receivedAveragePacketSize": 649.1468762126503,
      "relayed": 128,
      "relayedTotal": 1712,
      "relayedPerSecond": 127.97632437988945,
      "relayedBytes": 85932,
      "relayedTotalBytes": 1138757,
      "relayedBytesPerSecond": 85916.10552041141,
      "relayedAveragePacketSize": 665.1617990654206,
      "all": 363,
      "allTotal": 7350,
      "allPerSecond": 362.93285742109276,
      "allBytes": 224889,
      "allTotalBytes": 4648821,
      "allBytesPerSecond": 224847.40323022625,
      "allAveragePacketSize": 632.4926530612245,
      "relevant": 42,
      "relevantTotal": 1025,
      "relevantPerSecond": 41.992231437151226,
      "relevantBytes": 26776,
      "relevantTotalBytes": 646134,
      "relevantBytesPerSecond": 26771.047356218125,
      "relevantAveragePacketSize": 630.3746341463415,
      "duplicate": 15,
      "duplicateTotal": 493,
      "duplicatePerSecond": 14.997225513268296,
      "duplicateBytes": 11693,
      "duplicateTotalBytes": 382230,
      "duplicateBytesPerSecond": 11690.837195109745,
      "duplicateAveragePacketSize": 775.314401622718,
      "expired": 54,
      "expiredTotal": 1203,
      "expiredPerSecond": 53.990011847765864,
      "expiredBytes": 33916,
      "expiredTotalBytes": 741466,
      "expiredBytesPerSecond": 33909.726700533836,
      "expiredAveragePacketSize": 616.3474646716542,
      "connectedPeers": [
        "0x51777c0b8dba8b4dfe8a1c3d0a1edaa5b139b4e0",
        "0x252e15f6d77d75aad19e4ebb7b9ec6d5c89650a4",
        "0x23be90335e79d3245615985d91f69814f98a0fab",
        "0xea72ec663c69b5ddb0ee3b18c69b7ee9598c39a6"
      ],
      "knownPeersCount": 12,
      "position": [
        55.5,
        10.289119005203247,
        -39.20000076293945
      ]
    },
    "trackedPeers": [
      "ee262d",
      "8c39a6",
      "39b4e0"
    ],
    "visiblePeers": [
      
    ],
    "time": "2020-04-13T16:43:39.497Z",
    "sessionId": "338ada08-ced3-4a7c-b083-5aaa106f7d82"
  },
  "event": "Comms Status v2",
  "...": "..."
}
```